internal/host/conditions.go

```diff
@@ -0,0 +1,36 @@
+package host
```

Comment:
What's the difference between host conditions and host validations? Seems pretty much similar.

Reply:
conditions are hidden from the user, and they do not have an output that in the validations_info field.  They both can participate as conditions in the state machine.

Reply:
I think it could be simpler if all the validations/conditions would be in the same place and you will explicitly mention if you want it as a "hidden"/"soft"/etc validation that should affect/displayed to the user. Otherwise, I am afraid it might confuse contributors.

Reply:
So we will need additional flag for that.  I don't think flag is a good idea, and a structure that contains a function that is not used.  I am not for it.  It already exists in the cluster statemachine

Reply:
I think that `condition` vs. `validation` naming and implementation will cause contributors confusion and it should be avoided. Naming should be meaningful and currently, they mean the same thing.

In addition, there was a big effort for validations reporting (events, metrics) so even if all of this behavior happens in the background, there would still be required a mechanism to support this new reporting to analyze the behavior of conditions.

Therefore, I think validations and conditions should be merged. When a new contributor would want to add a new validation - he needs explicitly to choose whether the validation will be visible or hidden to the user.
Otherwise, a developer could mistakenly add a validation/condition just because that's what he is familiar with and not necessarily what he wished to happen.

@ronniel1 WDYT? Should "conditions" and "validations" be separated completely? Should we monitor and report conditions?

Reply:
I tend to agree with @YuviGold on this. However, seems to me that discussion has a wider scope than this PR, no?
Maybe we should also ask other people what they think about this and maybe a refactoring is due?
But since this issue has occurred before this PR, maybe this discussion should be done, and handled separately

Reply:
I still do not agree.  I think programming with flags is bad and this is what will happen if we merge validations and conditions.
Metrics and events are calculated upon validations_info which the conditions construct is not related to at all.

Reply:
@ori-amizur you are talking about how to implement what we are saying, and not if what we are saying is correct or not.
I assume we can find multiple ways to implement this, if we decide to go this way

Reply:
Will remove the conditions, and add these as condition functions in transition.go

Reply:
The difference is that condition can be a group of validations. 
We can probably write all the validations explicitly but maybe there is another way to handle it.
for example  generate the basic conditions

Reply:
I agree that our validations should have a format similar to conditions. so we don't need to implement it twice.